src/Simplify_Max.cpp

```diff
@@ -108,6 +122,16 @@ Expr Simplify::visit(const Max *op, ExprInfo *bounds) {
                rewrite(max(x, (x/c1)*c1 + c2), a, c1 > 0 && c2 <= 0) ||
                rewrite(max(((x + c0)/c1)*c1, x), b, c1 > 0 && c0 <= 0) ||
                rewrite(max(x, ((x + c0)/c1)*c1), a, c1 > 0 && c0 <= 0) ||
+
+               rewrite(max(x, min(x, y) + c0), a, c0 <= 0) ||
```

Comment:
Is there any reason for not including:
```
rewrite(max(x, c0 + min(x, y)), a, c0 <= 0) ||
rewrite(max(x, c0 + min(y, x)), a, c0 <= 0) ||
```
?

Reply:
The simplifier rewrites c0 + x to x + c0, so that LHS won't appear.